corehq/apps/reports/static/reports/js/bootstrap5/hq_report.js

```diff
@@ -21,14 +21,14 @@ hqDefine("reports/js/bootstrap5/hq_report", [
         self.datespan = options.datespan;
         self.filterSet = options.filterSet || false;
         self.needsFilters = options.needsFilters || false;
-        self.filterAccordion = options.filterAccordion || "#reportFilters";
-        self.filterSubmitSelector = options.filterSubmitSelector || '#paramSelectorForm button[type="submit"]';
+        self.filterAccordion = options.filterAccordion || "#reportFilters" + initialPageData.get('html_id_suffix');
+        self.filterSubmitSelector = options.filterSubmitSelector || '#paramSelectorForm' + initialPageData.get('html_id_suffix') + ' button[type="submit"]';
         self.filterSubmitButton = $(self.filterSubmitSelector);
-        self.toggleFiltersButton = options.toggleFiltersButton || "#toggle-report-filters";
-        self.exportReportButton = options.exportReportButton || "#export-report-excel";
-        self.emailReportButton = options.emailReportButton || "#email-report";
-        self.printReportButton = options.printReportButton || "#print-report";
-        self.emailReportModal = options.emailReportModal || "#email-report-modal";
+        self.toggleFiltersButton = options.toggleFiltersButton || "#toggle-report-filters" + initialPageData.get('html_id_suffix');
```

Comment:
Musing: I wish the existing code didn't rely so heavily on ids. I bet a lot of these are inside containers and instead of using id selectors, could do something like...
```
self.$accordion = "#reportFiltersAccordion";
self.$applyFilters = self.$accordion.find(".apply-filters");
```
...and in that situation, you'd add the suffix to the accordion selector but not the elements inside of it.

Reply:
Absolutely. I was hoping to use one ID to identify an outer element of a report, and then use `find()` to select the other elements we need inside it. Once the deadline that this work needs to meet has been met, this is one of the things I'd like to come back to.